app/controllers/billables_controller.rb

```diff
@@ -7,13 +7,12 @@ def index
   end
 
   def bill_entries
-    if params[:entries_to_bill]
-      params[:entries_to_bill].each do |value|
-        entry_type = value.split("-")[0]
-        entry_id = value.split("-")[1]
-        entry = to_object(entry_type).find(entry_id)
-        entry.update_attribute(:billed, true)
-      end
+    if params[:mileage_to_unbill] || params[:hour_to_unbill]
+      Mileage.where(id: params[:mileage_to_unbill]).update_all(billed: false)
+      Hour.where(id: params[:hour_to_unbill]).update_all(billed: false)
+    else
+      Mileage.where(id: params[:mileages_to_bill]).update_all(billed: true)
+      Hour.where(id: params[:hours_to_bill]).update_all(billed: true)
     end
```

Comment:
I think you should remove the if statements here, and in the view you should make a link_to instead of the button that does the unbilling of item and doesn't send the whole form